File: Delete nodes having greater value on right[29-August-2023]/Solution.py
#  >> PYTHON CODE <<

class Solution:
    def compute(self,head):
        def reverse(head):
            prev,next=None,None
            
            while(head!=None):
                next=head.next
                head.next=prev
                prev=head
                head=next
            
            return prev
        
        rev=reverse(head)
        
        ans=Node(-1)
        ptr=ans
        
        while(rev!=None):
            if(rev.data>=ptr.data):
                ptr.next=rev
                ptr=ptr.next
            rev=rev.next

        ptr.next=None
    
        return reverse(ans.next)

File: Delete nodes having greater value on right[29-August-2023]/solution.cpp
//  >> CPP CODE <<

class Solution
{
    public:
    Node* reverse(Node* head) {
        Node* prev = nullptr;
        Node* next = nullptr;
    
        while (head != nullptr) {
            next = head->next;
            head->next = prev;
            prev = head;
            head = next;
        }
    
        return prev;
    }
    
    Node* compute(Node* head) {
        Node* rev = reverse(head);
        Node* ans = new Node(-1);
        Node* ptr = ans;
    
        while (rev != nullptr) {
            if (rev->data >= ptr->data) {
                ptr->next = rev;
                ptr = ptr->next;
            }
    
            rev = rev->next;
        }
    
        ptr->next = nullptr;
    
        return reverse(ans->next);
    }
    
};